core/shared/src/main/scala/sigma/serialization/TypeSerializer.scala

```diff
@@ -189,7 +201,23 @@ class TypeSerializer {
         case SHeader.typeCode => SHeader
         case SPreHeader.typeCode => SPreHeader
         case SGlobal.typeCode => SGlobal
+        case SFunc.FuncTypeCode if VersionContext.current.isV6SoftForkActivated =>
+          val tdLength = r.getUByte()
+
+          val tDom = (1 to tdLength).map { _ =>
+            deserialize(r)
+          }
+          val tRange = deserialize(r)
+          val tpeParamsLength = r.getUByte()
+          val tpeParams = (1 to tpeParamsLength).map { _ =>
+            val ident = deserialize(r)
+            require(ident.isInstanceOf[STypeVar])
+            STypeParam(ident.asInstanceOf[STypeVar])
+          }
+          SFunc(tDom, tRange, tpeParams)
+        // todo: serialize tParams
         case _ =>
+          // todo: 6.0: replace 1008 check with identical but other opcode
```

Comment:
please clarify "identical but other".

Reply:
done, // todo: 6.0: replace 1008 check with identical behavior but other opcode, to activate
          //  ReplacedRule(1008 -> new opcode) during 6.0 activation